flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/util/KinesisConfigUtil.java

```diff
@@ -224,27 +222,19 @@ private static void validateOptionalPositiveDoubleProperty(Properties config, St
 		}
 	}
 
-	private static void validateOptionalDateProperty(Properties config, String timestampKey, String formatKey, String message) {
+	private static void validateOptionalDateProperty(Properties config, String timestampKey, String format, String message) {
 		if (config.containsKey(timestampKey)) {
-			if (config.containsKey(formatKey)) {
-				try {
-					SimpleDateFormat customDateFormat = new SimpleDateFormat(config.getProperty(formatKey));
-					customDateFormat.parse(config.getProperty(timestampKey));
-				} catch (ParseException | IllegalArgumentException | NullPointerException exception) {
-					throw new IllegalArgumentException(message);
-				}
-			} else {
+			try {
+				SimpleDateFormat customDateFormat = new SimpleDateFormat(format);
```

Comment:
IllegalArgumentException would be thrown. See `SimpleDateFormat` [API](https://docs.oracle.com/javase/7/docs/api/java/text/SimpleDateFormat.html#SimpleDateFormat(java.lang.String)).